cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/index.tsx

```diff
@@ -0,0 +1,203 @@
+
+const styles = (theme): StyleRules => {
+  return {
+    root: {
+      borderRight: `1px solid ${theme.palette.grey[300]}`,
+      height: '100%',
+    },
+    requestRow: {
+      padding: '10px',
+      lineHeight: '24px',
+      display: 'grid',
+      width: '100%',
+      gridTemplateColumns: '50px 1fr',
+      cursor: 'pointer',
+
+      '&:hover': {
+        backgroundColor: theme.palette.grey[700],
+      },
+    },
+    requestMethod: {
+      paddingLeft: '5px',
+      color: theme.palette.white[50],
+      width: '100%',
+      height: '100%',
+      fontWeight: 600,
+      display: 'flex',
+      justifyContent: 'center',
+      alignItems: 'flex-start',
+      fontSize: '10px',
+    },
+    requestMethodText: {
+      width: '100%',
+      textAlign: 'left',
+      alignSelf: 'center',
+    },
+    requestPath: {
+      width: `${LEFT_PANEL_WIDTH / 1.5}px`,
+      wordWrap: 'break-word',
+      textAlign: 'left',
+      textTransform: 'lowercase',
+      fontSize: '10px',
+      lineHeight: '1.3',
+    },
+    getMethod: {
+      color: theme.palette.green[50],
+    },
+    postMethod: {
+      color: theme.palette.orange[50],
+    },
+    putMethod: {
+      color: theme.palette.yellow[50],
+    },
+    deleteMethod: {
+      color: theme.palette.red[50],
+    },
+  };
+};
+
+enum RequestMethod {
+  GET = 'GET',
+  POST = 'POST',
+  PUT = 'PUT',
+  DELETE = 'DELETE',
+}
+
+interface IRequestHistory {
+  method: RequestMethod;
+  path: string;
+  body: string;
+  headers: {
+    pairs: [
+      {
+        key: string;
+        value: string;
+        uniqueId: string;
+      }
+    ];
+  };
+  response: string;
+  statusCode: number;
+}
+
+interface IRequestHistoryTabProps extends WithStyles<typeof styles> {
+  incomingRequest: boolean;
+  onRequestClick: (request: IRequestHistory) => void;
+  resetIncomingRequest: () => void;
+}
+
+const mapStateToProps = (state) => {
+  return {
+    incomingRequest: state.http.incomingRequest,
+  };
+};
+
+const mapDispatch = (dispatch) => {
+  return {
+    onRequestClick: (request: IRequestHistory) => {
+      dispatch({
+        type: HttpExecutorActions.setRequestHistoryView,
+        payload: request,
+      });
+    },
+    resetIncomingRequest: () => {
+      dispatch({
+        type: HttpExecutorActions.notifyIncomingRequest,
+        payload: {
+          incomingRequest: false,
+        },
+      });
+    },
+  };
+};
+
+const RequestHistoryTabView: React.FC<IRequestHistoryTabProps> = ({
+  classes,
+  incomingRequest,
+  onRequestClick,
+  resetIncomingRequest,
+}) => {
+  const [requestLog, setRequestLog] = React.useState(List<IRequestHistory>([]));
+
+  // Query through localstorage and popluate RequestHistoryTab
+  React.useEffect(() => {
+    let newRequestLog = List<IRequestHistory>([]);
+    Object.keys(localStorage).forEach((key) => {
+      if (key.startsWith('RequestHistory')) {
+        const newRequest = JSON.parse(localStorage.getItem(key));
```

Comment:
I think rather than saving each request log entry as its own item in the localStorage, it's better to store all the logs into a single key. You can construct an array of the history, then stringified the entire array